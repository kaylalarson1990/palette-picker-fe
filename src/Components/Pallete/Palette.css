.palette-box {
  align-items: center;
  display: flex;
  justify-content: space-between;
  width: 900px;
}

.box-color {
  border: 1px solid;
  margin: 3px 10px;
  height: 45px;
  width: 15%
}

.palette-title {
  width: 200px;
}

.delete-btn {
  cursor: pointer;
}