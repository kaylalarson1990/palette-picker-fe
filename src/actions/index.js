export const gatherProjects = projects => ({
  type: 'GATHER_PROJECTS',
  projects
})

export const gatherPalettes = palettes => ({
  type: 'GATHER_PALETTES',
  palettes
})

export const gatherProject = id => ({
  type: 'GATHER_PROJECT',
  id
})

export const gatherPalette = id => ({
  type: 'GATHER_PALETTE',
  id
})