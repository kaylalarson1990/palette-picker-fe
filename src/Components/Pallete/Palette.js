import React from 'react';
import {connect} from 'react-redux';
import './Palette.css'
import {gatherPalettes } from '../../actions';
import {fetchAllPalettes, deletePalette} from '../../apiCalls'

export const Palette = props => {
  const handleDelete = async id => {
    await deletePalette({id})
    let palettes = await fetchAllPalettes()
    props.gatherPalettes(palettes)
  }

  return (
    <div className='palette-box'>
      <h4 className='palette-title'>{props.palette_name}</h4>
      <article className='box-color' style={{backgroundColor: props.c1}}>
        <img />
      </article>
      <article className='box-color' style={{backgroundColor: props.c2}}>
        <img />
      </article>
      <article className='box-color' style={{backgroundColor: props.c3}}>
        <img />
      </article>
      <article className='box-color' style={{backgroundColor: props.c4}}>
        <img />
      </article>
      <article className='box-color' style={{backgroundColor: props.c5}}>
        <img />
      </article>
      <p className="delete-btn" onClick={e => handleDelete(props.id)}>‚ùå</p>
    </div>
  )
}


export const mapDispatchToProps = dispatch => ({
  gatherPalettes: palettes => dispatch(gatherPalettes(palettes))
})

export default connect(null, mapDispatchToProps)(Palette)