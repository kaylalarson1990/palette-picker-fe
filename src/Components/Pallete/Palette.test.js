import React from "react";
import { Palette, mapDispatchToProps } from "./Palette";
import { shallow } from "enzyme";
import { gatherPalettes } from "../../actions/index";
import "../../../setupTests";

describe("Projects", () => {
  let wrapper, instance, props;
  beforeEach(() => {
    props = {
      palettes: [
        {
          palette_name: "Test Palette",
          c1: "Red",
          c2: "Blue",
          c3: "Green",
          c4: "Yellow",
          c5: "White",
          id: 1
        }
      ]
    };
    wrapper = shallow(<Palette {...props} />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapDispatchToProps", () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.fn();
  });

  it("should dispatch gatherPalettes when handleSubmit is called", () => {
    const actionToDispatch = gatherPalettes({
      palettes: { palette_name: "Test" }
    });

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.gatherPalettes({
      palettes: { palette_name: "Test" }
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
