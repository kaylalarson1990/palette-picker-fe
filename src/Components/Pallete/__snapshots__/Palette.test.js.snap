// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot 1`] = `
<div
  className="palette-box"
>
  <h4
    className="palette-title"
  />
  <article
    className="box-color"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <img />
  </article>
  <article
    className="box-color"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <img />
  </article>
  <article
    className="box-color"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <img />
  </article>
  <article
    className="box-color"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <img />
  </article>
  <article
    className="box-color"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <img />
  </article>
  <p
    className="delete-btn"
    onClick={[Function]}
  >
    ‚ùå
  </p>
</div>
`;
